File: app/src/main/java/de/schooltec/datapass/WidgetIntentReceiver.java
package de.schooltec.datapass;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;

/**
 * Receiver getting triggered to update the widget manually on click.
 *
 * @author Martin Hellwig
 * @author Markus Hettig
 */
public class WidgetIntentReceiver extends BroadcastReceiver
{
    @Override
    public void onReceive(Context context, Intent intent)
    {
        new UpdateWidgetTask(intent.getIntExtra(UpdateWidgetTask.APP_WIDGET_ID, -1), context,
                UpdateWidgetTask.Mode.REGULAR, intent.getStringExtra(UpdateWidgetTask.APP_WIDGET_CARRIER))
                .executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
    }
}

File: app/src/main/java/de/schooltec/datapass/WidgetIntentReceiver.kt
package de.schooltec.datapass

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.AsyncTask

/**
 * Receiver getting triggered to update the widget manually on click.
 *
 * @author Martin Hellwig
 * @author Markus Hettig
 */
class WidgetIntentReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val updateWidgetTask = UpdateWidgetTask(
            intent.getIntExtra(UpdateWidgetTask.IDENTIFIER_APP_WIDGET_ID, -1),
            context,
            UpdateMode.REGULAR,
            intent.getStringExtra(UpdateWidgetTask.IDENTIFIER_APP_WIDGET_CARRIER)
        )
        updateWidgetTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
    }
}
